src/lib/kademlia/membership.ml

```diff
@@ -362,8 +362,7 @@ end = struct
       )
     in
     List.iter unbanned_lives ~f:(fun (peer, kkey) ->
-        let _ = Peer.Table.add ~key:peer ~data:kkey t.peers in
-        () ) ;
+        Peer.Table.set ~key:peer ~data:kkey t.peers ) ;
```

Comment:
This function is called when the Kad helper tells us there are new peers. If the supposedly new peer is already in the table somehow then we have a bug. I think it should crash in that case.

Reply:
so we are indeed getting an exception here. are you sure this means there is a bug? maybe we should just ignore this as we're ditching this code anyway?

```
File "lib/kademlia/membership.ml", line 605, characters 8-724: Membership threw
(monitor.ml.Error
  ("Hashtbl.add_exn got key already present"
    ((host 127.0.0.1) (discovery_port 3003) (communication_port 3002)))
  ("<backtrace elided in test>" "Caught by monitor block_on_async")).
  Raised at file "src/result.ml" (inlined), line 158, characters 17-26
  Called from file "src/thread_safe.ml", line 130, characters 29-63
  Called from file "lib/kademlia/membership.ml", line 607, characters 12-623
  Called from file "runtime-lib/runtime.ml", line 342, characters 8-12
  Re-raised at file "runtime-lib/runtime.ml", line 345, characters 6-13
  Called from file "runtime-lib/runtime.ml", line 358, characters 15-52
  Called from file "runtime-lib/runtime.ml", line 445, characters 52-83
  in TEST_MODULE at file "lib/kademlia/membership.ml", line 588, characters 4-1124: Mock Events
  in TEST_MODULE at file "lib/kademlia/membership.ml", line 485, characters 0-11495: Tests
```

Reply:
This is a test failure, yes?

Could it be caused by these duplicate lines in the test script:
```
    ; `On 3003
    ; `On 3003
```

Reply:
Alas that did not help. I'm changing it to `set` to get this in. If we think this is a significant bug that won't be obviated by @cmr 's stuff we can look into it